### ISSUE TITLE
CompareValidator client validation does not work with prefixed attributes

### ISSUE THREAD

**Issue by SilverFire** [yiisoft/validator#9]
In case of using form fields with prefixed attributes, the ID of input will be `model-prefix-attribute` (for example bellow `user-0-password`). During the generation of client validation JS in `CompareValidator`, the `compareAttribute` (which have to contain ID of target input) will be set to `model-password` instead of `model-0-password` (see `CompareValidator.php:219`), so the JS will generate false errors during the validation.

view.php:

``` php
use yii\widgets\ActiveForm;

$form = ActiveForm::begin();
foreach ($models as $i => $model) {
    print $form->field($model, "[$i]password");
    print $form->field($model, "[$i]confirm_password");
}
$form->end();
```

model.php:

``` php
class User extents Model {
...
    function rules() {
        return [
            [
                    ['password'],
                    'compare',
                    'compareAttribute' => 'confirm_password',
                    'on' => ['create'],
                ],
        ];
    }
...
}
```

Workaround: set `enableClientValidation` to `false` in model and use ajax validation with `ActiveForm::validateMultiple`

And I don't know, how to pass full attribute with prefix form `ActiveForm` to `CompareValidator`

**Comment by samdark** [comment]
Not going to implement client validation itself in this package.
